tests/python/test_fields_builder.py

```diff
@@ -6,171 +6,176 @@
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i
 
-    y = ti.field(ti.f32, [n])
-
     @ti.kernel
-    def func2():
-        for i in range(n):
+    def assign_field_multiple():
+        for i in range(shape):
             y[i] = i * 2
-        for i in range(n):
+        for i in range(shape):
             x[i] = i * 3
 
-    func2()
-
-    for i in range(n):
+    assign_field_multiple()
+    for i in range(shape):
         assert x[i] == i * 3
         assert y[i] == i * 2
 
-    func()
-
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i
 
 
 @ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan, ti.metal])
 def test_fields_builder_dense():
-    n = 5
-
+    shape = 5
     fb1 = ti.FieldsBuilder()
     x = ti.field(ti.f32)
-    fb1.dense(ti.i, n).place(x)
+    fb1.dense(ti.i, shape).place(x)
     fb1.finalize()
 
     @ti.kernel
-    def func1():
-        for i in range(n):
+    def assign_field_single():
+        for i in range(shape):
             x[i] = i * 3
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i * 3
 
     fb2 = ti.FieldsBuilder()
     y = ti.field(ti.f32)
-    fb2.dense(ti.i, n).place(y)
+    fb2.dense(ti.i, shape).place(y)
     z = ti.field(ti.f32)
-    fb2.dense(ti.i, n).place(z)
+    fb2.dense(ti.i, shape).place(z)
     fb2.finalize()
 
     @ti.kernel
-    def func2():
-        for i in range(n):
+    def assign_field_multiple():
+        for i in range(shape):
             x[i] = i * 2
-        for i in range(n):
+        for i in range(shape):
             y[i] = i + 5
-        for i in range(n):
+        for i in range(shape):
             z[i] = i + 10
 
-    func2()
-    for i in range(n):
+    assign_field_multiple()
+    for i in range(shape):
         assert x[i] == i * 2
         assert y[i] == i + 5
         assert z[i] == i + 10
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i * 3
 
 
 @ti.test(arch=[ti.cpu, ti.cuda, ti.metal])
 def test_fields_builder_pointer():
-    n = 5
-
+    shape = 5
     fb1 = ti.FieldsBuilder()
     x = ti.field(ti.f32)
-    fb1.pointer(ti.i, n).place(x)
+    fb1.pointer(ti.i, shape).place(x)
     fb1.finalize()
 
     @ti.kernel
-    def func1():
-        for i in range(n):
+    def assign_field_single():
+        for i in range(shape):
             x[i] = i * 3
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i * 3
 
     fb2 = ti.FieldsBuilder()
     y = ti.field(ti.f32)
-    fb2.pointer(ti.i, n).place(y)
+    fb2.pointer(ti.i, shape).place(y)
     z = ti.field(ti.f32)
-    fb2.pointer(ti.i, n).place(z)
+    fb2.pointer(ti.i, shape).place(z)
     fb2.finalize()
 
-    # test range-for
     @ti.kernel
-    def func2():
-        for i in range(n):
+    def assign_field_multiple0():
```

Comment:
if you want to change the name. Here better to be `assign_field_multiple_range_for` instead of `...0`. Likewise for the name below, better be struct_for

Reply:
A good advice! Fixed.